apis/logging/v1/input_source_types.go

```diff
@@ -0,0 +1,52 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// NOTE: The Enum validation on SourceSpec.Type must be updated if the list of types changes.
+
+// SourceSpec is a union of input source types.
+//
+// The fields of this struct define the set of known source types.
+type SourceSpec struct {
+	HTTP *HTTPSource `json:"http,omitempty"`
+}
+
+const (
+	FormatK8S  = "k8s"  // Log events in k8s list format, e.g. API audit log events.
+	FormatText = "text" // Newline-separated plain-text log records.
+)
+
+// HTTPSource receives encoded logs as a HTTP endpoint.
+type HTTPSource struct {
+
+	// FIXME TLS server configuration - can this be handled in the Service,
+	// or do we need vector to have certificates? Can the service be bypassed?
+
+	// FIXME What to do about service name clashes? Can't generate random name, service name is used to locate the source.
+
+	// Name of the source.
+	// A Service resource is generated for each source,
+	// the name must be a unique Service name in the ClusterLogForwarder's namespace.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// Port that this source will listen on.
+	// Used to create a Service for this port.
+	//
+	// +required
+	Port corev1.ServicePort `json:"port"`
```

Comment:
Do we really want this to be `corev1.ServicePort` ?  Does an admin need to really be capable of defining all the fields of ServicePort?  Aren't we in control meaning we will create the service that fronts the pod so we can only need them to find the outward facing service port and we control the rest

Reply:
I was thinking that this _is_ the outward facing service port, we would create a service with this port for the user. We are in control of how that service is linked to the Pod. If you have an alternative in mind I'd like to see it, I may be off in the weeds here.

Reply:
My implication was that " corev1.ServicePort" is more then an integer for the port number which opens up the API for additional configuration.  Do we need all that config or can we default to someone providing a port number? Can we default a port number and make an optional override?

Reply:
willing to review, lets continue on https://github.com/openshift/cluster-logging-operator/pull/2155